### PR TITLE
Add GCP Support

### PR DESCRIPTION
**Pull request by dchukhin**
This pull request adds support for GCP in `providers.gcp`, similarly to the functions in `providers.aws`.

### CODE REVIEW

File: kubesae/providers/gcp.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        repository: Name of docker repository, e.g. us.gcr.io/myproject/myproject
+    """
+    registry = c.config.repository.split("/")[0]
+    c.run("gcloud auth login")
+    c.run(f"gcloud config set project {c.config.app}")
+    c.run(f"gcloud auth configure-docker {registry}")
+
+
+@invoke.task()
+def configure_kubeconfig(c, cluster=None, region=None):
```
Comment (dpoirier) [inline kubesae/providers/gcp.py]: Can this be named ``configure-gcp-kubeconfig`` for consistency with ``configure-eks-kubeconfig``?

**Review by dpoirier — COMMENTED** [review]
Didn't really review, just had one naming suggestion.

File: kubesae/providers/gcp.py (hunk 1 above)
Comment (dchukhin) [reply kubesae/providers/gcp.py]: Sure, that can be done

File: kubesae/providers/gcp.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    c.run(f"gcloud config set project {c.config.app}")
+    c.run(f"gcloud auth configure-docker {registry}")
+
+
+@invoke.task()
+def configure_gcp_kubeconfig(c, cluster=None, region=None):
+    """
+    Authenticate into GCP to get credentials for the cluster.
+
+    Usage: inv gcp.configure-kubeconfig --cluster=<CLUSTER> --region=<REGION>
```
Comment (vkurup) [inline kubesae/providers/gcp.py]: s/configure-kubeconfig/configure-gcp-kubeconfig/

**Review by vkurup — APPROVED** [review]
Let's add these tasks to the README (in the 'Task reference' section). Also, in the README, it would be nice to change the paragraph starting with "The following code snippet imports all of the the current collections," to instead state that the following code snippet shows an example for AWS, but that GCP is also supported.

**Pull request merged**